test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/VSRestoreSettingsUtilityTests.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit;
+using NuGet.ProjectModel;
+using NuGet.Commands;
+using System.Linq;
+using NuGet.Configuration;
+using NuGet.Test.Utility;
+using System.Collections.Generic;
+using Xunit.Extensions;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    public class VSRestoreSettingsUtilityTests
+    {
+        [Theory]
+        [MemberData(nameof(GetVSRestoreSettingsUtilities_RestoreSourceData))]
```

Comment:
InlineData has a limitation in that it only accepts constants. In this case I wanted to pass a new string array and that just doesn't work.

Using member data, you pass an enumerable and xunit iterates over that enumerable and tries to match the types in the object array with the arguments of the method.
Each element of the IEnumerable is a separate tests, so in this case, 5 tests will show up.